src/backend/env/mod.rs

```diff
@@ -1083,14 +1083,22 @@ impl State {
             .users
             .values()
             .flat_map(|user| user.mintable_tokens(self, user_shares, boostraping_mode))
-            .filter(|(_, balance)| *balance > 0)
```

Comment:
I think it was a good optimization to avoid creating a new entry in the hash table for users that don't get any tokens.

Maybe even better optimization would be to adjust `mintable_tokens` to never return such entries.

Reply:
Done!